src/tokenizedSubaccounts/LevBasisTSA.sol

```diff
@@ -0,0 +1,476 @@
+import {IOptionAsset} from "v2-core/src/interfaces/IOptionAsset.sol";
+import {ISpotFeed} from "v2-core/src/interfaces/ISpotFeed.sol";
+import {IPerpAsset} from "v2-core/src/interfaces/IPerpAsset.sol";
+import {IDepositModule} from "../interfaces/IDepositModule.sol";
+import {IWithdrawalModule} from "../interfaces/IWithdrawalModule.sol";
+import {IMatching} from "../interfaces/IMatching.sol";
+import {IRfqModule} from "../interfaces/IRfqModule.sol";
+import {IWrappedERC20Asset} from "v2-core/src/interfaces/IWrappedERC20Asset.sol";
+
+import {
+  StandardManager, IStandardManager, IVolFeed, IForwardFeed
+} from "v2-core/src/risk-managers/StandardManager.sol";
+import {ITradeModule} from "../interfaces/ITradeModule.sol";
+import {CollateralManagementTSA} from "./CollateralManagementTSA.sol";
+
+/// @title LeveragedBasisTSA
+/// @notice A TSA that accepts a base asset, borrows against it to buy more, and then opens short perps to neutralise
+/// the delta of the borrowed portion, earning the perp funding in the process.
+///
+/// decision to borrow is based on current perp funding rate and usdc borrow rate. Only borrow more if the perp funding
+/// is higher than the borrow rate by a larger degree.
+///
+/// only closes the position if the borrow rate is higher than the perp funding rate.
+contract LeveragedBasisTSA is CollateralManagementTSA {
+  using IntLib for int;
+  using SafeCast for int;
+  using SafeCast for uint;
+  using DecimalMath for uint;
+  using SignedDecimalMath for int;
+
+  struct LBTSAInitParams {
+    ISpotFeed baseFeed;
+    IDepositModule depositModule;
+    IWithdrawalModule withdrawalModule;
+    ITradeModule tradeModule;
+    IPerpAsset perpAsset;
+  }
+
+  struct LBTSAParams {
+    ///////
+    // Guardrail params
+    ///////
+    // Trade bounds
+    uint maxPerpFee;
+    /// @dev Max loss per unit of base traded, in base
+    int maxBaseLossPerBase;
+    /// @dev Max loss per unit of perp traded, in base
+    int maxBaseLossPerPerp;
+    // Delta
+    int deltaTarget;
+    int deltaTargetTolerance;
+    // Leverage
+    uint leverageFloor;
+    uint leverageCeil;
+    // EMA
+    /// @dev Factor for half life of EMA decay (e.g. 0.0002 ~= 1hr)
+    uint emaDecayFactor;
+    /// @dev Target % of of tvl that we can maximally lose via EMA calculation
+    uint markLossEmaTarget;
+    uint minSignatureExpiry;
+    uint maxSignatureExpiry;
+  }
+
+  /// @custom:storage-location erc7201:lyra.storage.LeveragedBasisTSA
+  struct LBTSAStorage {
+    IDepositModule depositModule;
+    IWithdrawalModule withdrawalModule;
+    ITradeModule tradeModule;
+    IRfqModule rfqModule;
+    IPerpAsset perpAsset;
+    ISpotFeed baseFeed;
+    LBTSAParams lbParams;
+    CollateralManagementParams collateralManagementParams;
+    int markLossEma;
+    uint markLossLastTs;
+    /// @dev Only one hash is considered valid at a time, and it is revoked when a new one comes in.
```

Comment:
note: might be worth adding that:
- the vault can still quote several orders as long as the signatures are checked via `static call`
- if there is a partial fill, a new order would invalidate the remaining portion of the partial fill. 

I'm just worried we might forget this nuance next time we revisit this contract.